src/types/mod.rs

```diff
@@ -60,22 +60,13 @@ pub struct Message<T: Clone> {
 }
 
 impl<T: Clone> Message<T> {
-    pub fn new(
-        partition: Partition,
-        offset: u64,
-        payload: T,
-        timestamp: DateTime<Utc>,
-        next_offset: Option<u64>,
-    ) -> Self {
+    pub fn new(partition: Partition, offset: u64, payload: T, timestamp: DateTime<Utc>) -> Self {
         Self {
             partition,
             offset,
             payload,
             timestamp,
-            next_offset: match next_offset {
-                Some(v) => v,
-                None => offset + 1,
-            },
+            next_offset: offset + 1,
```

Comment:
We don't need to store it. That would be a breaking change in the Python library, but I think it is worthwhile.

Reply:
Here is the Python version of this change: https://github.com/getsentry/arroyo/pull/73.

Small difference - I kept it as a computed property over there to be more compatible with the prior version of the library, but just a function here as that doesn't exist in Rust.

Reply:
Well, I think I found the likely reason next_offset was originally passed in. It's because of the way the file storage backend works. In that backend, the next_offset is something other than offset + 1